storage_service/locations/api/resources.py

```diff
@@ -839,6 +840,22 @@ def aip_store_callback_request(self, request, bundle, **kwargs):
         else:
             return http.HttpNoContent()
 
+    @_custom_endpoint(expected_methods=['post'])
+    def reindex_request(self, request, bundle, **kwargs):
+        package = bundle.obj
+        if package.package_type != Package.TRANSFER:
+            return http.HttpBadRequest(json.dumps({"error": True, "message": "This package is not a transfer."}), content_type='application/json')
+        if package.current_location.purpose != Location.BACKLOG:
+            return http.HttpBadRequest(json.dumps({"error": True, "message": "This package is not part of a transfer backlog location."}), content_type='application/json')
```

Comment:
Also, is it possible for SS to know about a transfer and yet not have that transfer be in backlog? I tried to make a request that would trigger returning of the "This package is not part of a transfer backlog location." message but was not able to. I'm not saying this check should be removed; I'm just wondering if and how it's possible to trigger this response.